functions/exercises.py

```diff
@@ -333,6 +422,10 @@ def squared_nums(without_odds=False):
     :param without_odds: If True Odd numbers are not returned
     :return:
     """
+    if without_odds:
+        return [n**2 for n in range(1,31) if n%2==0]
+    else:
+        return[n**2 for n in range(1,31)]

```

Comment:
```suggestion
    sqr_nums = [n**2 for n in range(1, 31)]

    if without_odds:
        return list(filter(lambda x: n%2==0), sqr_nums))

    return sqr_nums
```
a bit simpler and without code duplication